### PR TITLE
chore(server): link parts to bikes

### PR DESCRIPTION
**Pull request by Mackbellemore**
#

## Changes this PR introduces

- When you create a bike now, you need to pass in all the ids of the parts you would like to use to assemble it and it will decrement the stock on parts and materials accordingly. I did not implement transactions since we need to have replica sets to be able to do transactions and dat shit is a pain.

## How to test this PR
payload example: you can use this exact one to test POST /bikes endpoint
```js
{
	"description": "carbon road bike",
	"color": "blue",
	"parts": {
		"handle_bar": "60503e2615f060001ad034f2",
		"wheels": "60503e2615f060001ad034f8",
		"chain": "60503e2615f060001ad034f3",
		"frame": "60503e2615f060001ad034f7",
		"brakes": "60503e2615f060001ad034f4",
		"fork": "6050344e9c4b6428e2ab0f7d",
		"seat": "60503e2615f060001ad034f6",
		"pedal": "60503e2615f060001ad034f5"
	},
	"name": "test-bike",
	"stock": 1,
	"sellingPrice": 10,
	"costPrice": 10,
	"weightType": "lb",
	"weightAmount": 10
}
```

## Notes for reviewers

- a note

## Checklist

- [x] Have I renamed the title of this pull request to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)?
- [x] I will squash and merge when my PR is approved.
- [x] I have added tests to cover my changes (if applicable).

### CODE REVIEW

**Comment by Mackbellemore** [comment]
Task linked: [CU-npzdz3 Link bikes, parts and materials](https://app.clickup.com/t/npzdz3)

File: server/app/services/BikeService.ts [hunk 1]
```diff
@@ -42,4 +49,23 @@ export class BikeService {
 
     return bike;
   }
+
+  private async handlePartStock(body: IBike): Promise<void> {
+    const callStack = [];
+    const { stock: bikeStock } = body;
+    for (const [partName, partId] of Object.entries(body.parts)) {
+      // weird type converting here to escape the optional array return type
+      const { stock: partStock, name } = (await (this.partService.get(partId) as unknown)) as IPart;
```
Comment (gordonpn) [inline server/app/services/BikeService.ts:58]: Just TypeScript tings 🤨

File: server/app/services/BikeService.ts [hunk 2]
```diff
@@ -2,17 +2,24 @@ import { inject, injectable } from 'inversify';
...
+    @inject(TYPES.PartService) private partService: PartService
+  ) {}
 
   public async getBikes(): Promise<IBike[]> {
     return this.bikeRepo.getList();
   }
 
   public async createBike(body: IBike): Promise<IBike> {
+    // one day we can have some nice transactions, but not today
+    await this.handlePartStock(body);
```
Comment (gordonpn) [inline server/app/services/BikeService.ts:22]: 💀

File: .docker/docker-entrypoint-initdb.d/1-init.js [hunk 3]
```diff
@@ -38,43 +41,102 @@ db.materials.insert([
...
+  },
+  {
+    name: "fast-wheels",
+    quality: "aluminium",
+    type: "wheels",
+    color: "red",
+    stock: 1000,
+  },
+  {
+    name: "drip",
```
Comment (gordonpn) [inline .docker/docker-entrypoint-initdb.d/1-init.js:101]: image

**Review by gordonpn — APPROVED** [review]

File: .docker/docker-entrypoint-initdb.d/1-init.js (hunk 3 above)
Comment (Mackbellemore) [reply .docker/docker-entrypoint-initdb.d/1-init.js:101]: its an aluminium chain, max drip

**Review by tiffzeng — APPROVED** [review]

**Pull request merged**